scheduler/src/cook/kubernetes/api.clj

```diff
@@ -488,8 +513,12 @@
                                                                           convert-resource-map))
                                                                 containers)
                                         resource-map (apply merge-with + container-requests)
-                                        gpu-model (some-> pod .getSpec .getNodeSelector (get "cloud.google.com/gke-accelerator"))]
-                                    (add-gpu-model-to-resource-map gpu-model resource-map))))
+                                        ^V1NodeSelector nodeSelector (some-> pod .getSpec .getNodeSelector)
+                                        gpu-model (some-> nodeSelector (get "cloud.google.com/gke-accelerator"))
+                                        disk-type (some-> nodeSelector (get (pool->disk-type-label-name pool-name)))
+                                        update-gpu-in-res-map (fn [res-map] (add-gpu-model-to-resource-map gpu-model res-map))
```

Comment:
Instead of having update-gpu-in-res-map and update-disk-in-res-map you could replace:

(-> resource-map update-gpu-in-res-map update-disk-in-res-map)

with

(->> resource-map
  (add-gpu-model-to-resource-map gpu-model)
  (add-disk-type-to-resource-map disk-type))

Reply:
done, thanks for the suggestion!!